geode-core/src/test/java/org/apache/geode/management/internal/cli/functions/DestroyAsyncEventQueueFunctionTest.java

```diff
@@ -0,0 +1,140 @@
+
+  private static final String TEST_AEQ_ID = "Test-AEQ";
+  private AsyncEventQueue mockAEQ;
+  private FunctionContext mockContext;
+  private DestroyAsyncEventQueueFunctionArgs mockArgs;
+  private GemFireCacheImpl cache;
+
+  @Before
+  public void setUp() throws Exception {
+    mockAEQ = mock(AsyncEventQueueImpl.class);
+    mockContext = mock(FunctionContext.class);
+    mockArgs = mock(DestroyAsyncEventQueueFunctionArgs.class);
+    cache = Fakes.cache();
+
+    when(mockArgs.getId()).thenReturn(TEST_AEQ_ID);
+    when(mockAEQ.getId()).thenReturn(TEST_AEQ_ID);
+  }
+
+  @Test
+  public void execute_validAeqId_OK() throws Throwable {
+    XmlEntity xmlEntity = mock(XmlEntity.class);
+    DestroyAsyncEventQueueFunction spyFunction = spy(DestroyAsyncEventQueueFunction.class);
+    doReturn(xmlEntity).when(spyFunction).getAEQXmlEntity(anyString(), anyString());
+    when(cache.getAsyncEventQueue(TEST_AEQ_ID)).thenReturn(mockAEQ);
+
+    TestResultSender resultSender = new TestResultSender();
+    FunctionContext context = new FunctionContextImpl(cache, "functionId", mockArgs, resultSender);
+    spyFunction.execute(context);
+
+    List<?> results = resultSender.getResults();
+    assertThat(results.size()).isEqualTo(1);
+    CliFunctionResult result = (CliFunctionResult) results.get(0);
+    assertThat(result.isSuccessful()).isTrue();
+    assertThat(result.getXmlEntity()).isNotNull();
+    assertThat(result.getThrowable()).isNull();
+  }
+
+  @Test
+  public void execute_nonexistentAeqId_returnsError() throws Throwable {
+    when(cache.getAsyncEventQueue(TEST_AEQ_ID)).thenReturn(null);
+
+    TestResultSender resultSender = new TestResultSender();
+
+    FunctionContext context = new FunctionContextImpl(cache, "functionId", mockArgs, resultSender);
+    new DestroyAsyncEventQueueFunction().execute(context);
+    List<?> results = resultSender.getResults();
+    assertThat(results.size()).isEqualTo(1);
+    CliFunctionResult result = (CliFunctionResult) results.get(0);
+    assertThat(result.isSuccessful()).isFalse();
+    assertThat(result.getMessage()).containsPattern(TEST_AEQ_ID + ".*not found");
+  }
+
+  @Test
+  public void execute_nonexistentAeqIdIfExists_returnsSuccess() throws Throwable {
+    when(cache.getAsyncEventQueue(TEST_AEQ_ID)).thenReturn(null);
+    when(mockArgs.isIfExists()).thenReturn(true);
+
+    TestResultSender resultSender = new TestResultSender();
+
+    FunctionContext context = new FunctionContextImpl(cache, "functionId", mockArgs, resultSender);
+    new DestroyAsyncEventQueueFunction().execute(context);
+    List<?> results = resultSender.getResults();
+    assertThat(results.size()).isEqualTo(1);
+    CliFunctionResult result = (CliFunctionResult) results.get(0);
+    assertThat(result.isSuccessful()).isTrue();
+    assertThat(result.getMessage()).containsPattern("Skipping:.*" + TEST_AEQ_ID + ".*not found");
+  }
+
+  private static class TestResultSender implements ResultSender {
```

Comment:
You can use a mock resultsender and argument capture to get the results instead of creating an implementation yourself. This is OK too, but more test code than necessary. (not a must fix, but if you need help converting it, I can pair with you).

Reply:
I like this suggestion.